### PR TITLE
CHE-5721: Importing java formatter for the project or workspace

### PR DESCRIPTION
**Pull request by svor**
### What does this PR do?
Provides an ability to import eclipse java formatter. The formatter can be applied for the custom project or for the whole workspace.
selection_012

Simple demo about how it works: https://youtu.be/oxi5OOvJsPQ

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/5721

#### Changelog
Importing Eclipse formatter for the java classes

#### Release Notes
Importing Eclipse formatter for the java classes

### CODE REVIEW

**Comment by svor** [comment]
ci-test

**Review by evidolob — APPROVED** [review]

**Review by vparfonov — APPROVED** [review]

**Comment by musienko-maxim** [comment]
The stable part of selenium tests passed as expected

**Review by musienko-maxim — APPROVED** [review]

File: plugins/plugin-java/che-plugin-java-ext-jdt/org-eclipse-jdt-ui/src/main/java/org/eclipse/jdt/internal/corext/format/CheCodeFormatterOptions.java [hunk 1]
```diff
@@ -37,6 +39,18 @@
     return DefaultCodeFormatterConstants.getEclipseDefaultSettings();
   }
 
+  public static Map<String, String> getFormatSettingsFromFile(File file) {
+    SAXParserFactory factory = SAXParserFactory.newInstance();
+    XMLParser parserXML = new XMLParser();
+    try {
+      SAXParser parser = factory.newSAXParser();
+      parser.parse(new FileInputStream(file), parserXML);
```
Comment (skabashnyuk) [inline plugins/plugin-java/che-plugin-java-ext-jdt/org-eclipse-jdt-ui/src/main/java/org/eclipse/jdt/internal/corext/format/CheCodeFormatterOptions.java]: Stream closing?

**Review by skabashnyuk — CHANGES_REQUESTED** [review]

File: plugins/plugin-java/che-plugin-java-ext-jdt/org-eclipse-jdt-ui/src/main/java/org/eclipse/jdt/internal/corext/format/CheCodeFormatterOptions.java (hunk 1 above)
Comment (svor) [reply plugins/plugin-java/che-plugin-java-ext-jdt/org-eclipse-jdt-ui/src/main/java/org/eclipse/jdt/internal/corext/format/CheCodeFormatterOptions.java]: fixed, thank you!

**Comment by svor** [comment]
@slemeur please take a look

**Review by skabashnyuk — APPROVED** [review]

**Comment by slemeur** [comment]
Wait wait wait - where is the release note and the documentation for that new feature?

**Comment by slemeur** [comment]
@vparfonov ?

**Pull request merged**